lib/asyncapiSchemaFormatParser.js

```diff
@@ -0,0 +1,46 @@
+const Ajv = require('ajv');
+const ParserError = require('./errors/parser-error');
+const asyncapi = require('@asyncapi/specs');
+const { improveAjvErrors } = require('./utils');
+
+module.exports = {
+  parse,
+  getMimeTypes
+};
+
+async function parse({ message, oryginalAsyncAPIDocument, fileFormat, parsedAsyncAPIDocument }) {
+  const ajv = new Ajv({
+    jsonPointers: true,
+    allErrors: true,
+    schemaId: 'id',
+    logger: false,
+  });
+  const payloadSchema = preparePayloadSchema(asyncapi[parsedAsyncAPIDocument.asyncapi]);
+  const validate = ajv.compile(payloadSchema);
```

Comment:
It would be great to move these lines outside the `parse` function. Compiling the schema is a relatively slow process and just needs to be done once per the AsyncAPI version. If we leave it here, it's compiling the schema on every `parse` call.

Another solution would be to cache the `validate` variable so it's only compiled once. For instance:

```js
let compiledSchemasCache = new Map();

async function parse({ message, originalAsyncAPIDocument, fileFormat, parsedAsyncAPIDocument }) {
  let validate;
  if (compiledSchemasCache.get(parsedAsyncAPIDocument.asyncapi)) {
    validate = compiledSchemasCache.get(parsedAsyncAPIDocument.asyncapi);
  } else {
    const ajv = new Ajv({
      jsonPointers: true,
      allErrors: true,
      schemaId: 'id',
      logger: false,
    });
    const payloadSchema = preparePayloadSchema(asyncapi[parsedAsyncAPIDocument.asyncapi]);
    validate = ajv.compile(payloadSchema);
    compiledSchemasCache.set(parsedAsyncAPIDocument.asyncapi, validate);
  }

  const valid = validate(message.payload);
  ...
}
```

Reply:
@fmvilas I do not think we have to worry as according to readme, ajv already caches it for us:
> Ajv compiles schemas to functions and caches them in all cases (using schema serialized with fast-json-stable-stringify or a custom function as a key), so that the next time the same schema is used (not necessarily the same object instance) it won't be compiled again.

Reply:
Oh! I didn't know they added this. That's great then. Forget about it.